pysmt/optimization/optimsat.py

```diff
@@ -106,15 +106,15 @@ def _assert_msat_goal(self, goal):
             cost_function = goal.term()
             obj_fun = self.converter.convert(cost_function)
 
-        else:
+        else: #elif maxsmt
             raise GoalNotSupportedError("optimathsat", goal)
 
-        msat_obj = make_fun(self.msat_env(), obj_fun, False)
+        msat_obj = make_fun(self.msat_env(), obj_fun, signed = goal.signed())
         self._msat_lib.msat_assert_objective(self.msat_env(), msat_obj)
         return msat_obj
 
     def optimize(self, goal, **kwargs):
-
+        #if goal esiste fai altrimenti ...
```

Comment:
Please check for comments in Italian and remove/translate them.

Also, this comment is quite useless IMHO...